statsmodels/tsa/stattools.py

```diff
@@ -1735,6 +1735,282 @@ def _sigma_est_kpss(resids, nobs, lags):
+        Zivot-Andrews unit-root test.
+
+        Notes
+        -----
+        The p-values are generated through Monte Carlo simulation using 100,000
+        replications and 2000 data points.
+        """
+        self._za_critical_values = {}
+        # constant-only model
+        self._c = ((0.001, -6.78442), (0.100, -5.83192), (0.200, -5.68139),
+             (0.300, -5.58461), (0.400, -5.51308), (0.500, -5.45043),
+             (0.600, -5.39924), (0.700, -5.36023), (0.800, -5.33219),
+             (0.900, -5.30294), (1.000, -5.27644), (2.500, -5.03340),
+             (5.000, -4.81067), (7.500, -4.67636), (10.000, -4.56618),
+             (12.500, -4.48130), (15.000, -4.40507), (17.500, -4.33947),
+             (20.000, -4.28155), (22.500, -4.22683), (25.000, -4.17830),
+             (27.500, -4.13101), (30.000, -4.08586), (32.500, -4.04455),
+             (35.000, -4.00380), (37.500, -3.96144), (40.000, -3.92078),
+             (42.500, -3.88178), (45.000, -3.84503), (47.500, -3.80549),
+             (50.000, -3.77031), (52.500, -3.73209), (55.000, -3.69600),
+             (57.500, -3.65985), (60.000, -3.62126), (65.000, -3.54580),
+             (70.000, -3.46848), (75.000, -3.38533), (80.000, -3.29112),
+             (85.000, -3.17832), (90.000, -3.04165), (92.500, -2.95146),
+             (95.000, -2.83179), (96.000, -2.76465), (97.000, -2.68624),
+             (98.000, -2.57884), (99.000, -2.40044), (99.900, -1.88932), )
+        self._za_critical_values['c'] = np.asarray(self._c)
+        # trend-only model
+        self._t = ((0.001, -83.9094), (0.100, -13.8837), (0.200, -9.13205),
+             (0.300, -6.32564), (0.400, -5.60803), (0.500, -5.38794),
+             (0.600, -5.26585), (0.700, -5.18734), (0.800, -5.12756),
+             (0.900, -5.07984), (1.000, -5.03421), (2.500, -4.65634),
+             (5.000, -4.40580), (7.500, -4.25214), (10.000, -4.13678),
+             (12.500, -4.03765), (15.000, -3.95185), (17.500, -3.87945),
+             (20.000, -3.81295), (22.500, -3.75273), (25.000, -3.69836),
+             (27.500, -3.64785), (30.000, -3.59819), (32.500, -3.55146),
+             (35.000, -3.50522), (37.500, -3.45987), (40.000, -3.41672),
+             (42.500, -3.37465), (45.000, -3.33394), (47.500, -3.29393),
+             (50.000, -3.25316), (52.500, -3.21244), (55.000, -3.17124),
+             (57.500, -3.13211), (60.000, -3.09204), (65.000, -3.01135),
+             (70.000, -2.92897), (75.000, -2.83614), (80.000, -2.73893),
+             (85.000, -2.62840), (90.000, -2.49611), (92.500, -2.41337),
+             (95.000, -2.30820), (96.000, -2.25797), (97.000, -2.19648),
+             (98.000, -2.11320), (99.000, -1.99138), (99.900, -1.67466), )
+        self._za_critical_values['t'] = np.asarray(self._t)
+        # constant + trend model
+        self._ct = ((0.001, -38.17800), (0.100, -6.43107), (0.200, -6.07279),
+              (0.300, -5.95496), (0.400, -5.86254), (0.500, -5.77081),
```

Comment:
check indentation.  if you're not familiar with the convention, you can run `flake8 --isolated statsmodels/tsa/stattools.py` and ignore all the alerts for the rest of the module.

Reply:
are these actually needed as attributes of `self`?

If these attributes are identical across all instances and never edited, no need to bother with `__init__`, just put them as class attributes

Reply:
1) I fixed the indentation although there seems to be some ambiguity between the different style parsers. The form I used (which seems to be the most preferred) is:
list = (
  (x1,y1), ...
  (x2,y2), ...
)

2) I originally used __init__ for the CV list specifically in case there is ever a desire/need to split out the routine as a base class then inherit for different distributions. The question of class attribute vs object attribute is not too important here since the use model (in a single-threaded context) is to instantiate once then re-use the object for multiple runs as opposed to instantiating a new object for each run. I left it as is for now but can certainly update if there is a strong preference.